torch_geometric/loader/neighbor_loader.py

```diff
@@ -197,17 +211,13 @@ def _hetero_sparse_neighbor_sample(self, index_dict: Dict[str, Tensor]):
             )
         return node_dict, row_dict, col_dict, edge_dict
 
-    def __call__(self, index: Union[List[int], Tensor]):
-        if not isinstance(index, torch.LongTensor):
-            index = torch.LongTensor(index)
+    def __call__(self, index: Union[List[int], Tensor, HeteroNodeList]):
```

Comment:
@wsad1 Can I get your opinion on this (note the PR is overall still somewhat WIP).

I was finding the logic here getting quite complicated. Do you have any suggestions for simplification? One thing I was considering was providing a custom collate_fn that handles the conversion to some container like 
```
class HeteroNodes:
    ...
```
instead of doing all the type inference here.

Reply:
Thanks for the update. I'll check it out once I am back from a short vacation on Tuesday morning.

Reply:
No rush, thank you.